### PR TITLE
feat(input text): input component

### PR DESCRIPTION
**Pull request by abdelazizsakhi**

### CODE REVIEW

File: libs/ngx-dsfr/input/src/lib/testing/input.harness.ts [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+ * To see more : https://material.angular.io/cdk/test-harnesses/overview
+ */
+export class DsfrInputHarness extends ComponentHarness {
+  static hostSelector: string = 'dsfr-input';
+
+  private getInputElement: AsyncFactoryFn<TestElement> = this.locatorFor('input[type=text]');
+
+  private async getInputAttribute(name: string): Promise<(string | null)> {
+    const input: TestElement = await this.getInputElement();
+    const attributes = await input.getAttribute(name);
```
Comment (flowg) [inline libs/ngx-dsfr/input/src/lib/testing/input.harness.ts:24]: `attribute` without an 's' seems more appropriate, don't you think ?

File: libs/ngx-dsfr/input/src/lib/testing/input.harness.ts [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+   */
+  async getInputValue(): Promise<(string | null)> {
+    return this.getInputAttribute('value');
+  }
+
+  /**
+   * Retrieves the placeholder value of the input tags
+   * in the DsfrInputComponent's Template
+   *
+   * @returns A Promise that resolves to the name values as a strings or null
```
Comment (flowg) [inline libs/ngx-dsfr/input/src/lib/testing/input.harness.ts:53]: For all the comments : it's `input tag` and `value as a string`

File: libs/ngx-dsfr/input/src/lib/testing/input.component.spec.ts [hunk 3]
```diff
@@ -0,0 +1,87 @@
...
+import { TestHostComponent } from './test-host.component';
+import { DsfrInputHarness } from './input.harness';
+import { FormsModule } from '@angular/forms';
+
+describe('DsfrInputComponent', () => {
+  let fixture: ComponentFixture<TestHostComponent>;
+  let testHost: TestHostComponent;
+  let componentUnderTest: DsfrInputComponent;
+  let harnessLoader: HarnessLoader;
+  let dsfrRadioHarness: DsfrInputHarness;
```
Comment (flowg) [inline libs/ngx-dsfr/input/src/lib/testing/input.component.spec.ts:22]: `dsfrInputHarness`

File: libs/ngx-dsfr/input/src/lib/testing/input.component.spec.ts [hunk 4]
```diff
@@ -0,0 +1,87 @@
...
+  });
+
+  it('should throw an error when no id is provided', async () => {
+    try {
+      fixture.detectChanges();
+      throw 'It should have thrown an error about "ID"';
+    } catch (error) {
+      expect(error).toBe(EMPTY_ID_ERROR);
+    }
+  });
```
Comment (flowg) [inline libs/ngx-dsfr/input/src/lib/testing/input.component.spec.ts:65]: Can be easily adapted to add the same check for `label` once `Id` is provided

File: libs/ngx-dsfr/input/src/lib/input.component.ts [hunk 5]
```diff
@@ -0,0 +1,148 @@
...
+    this.setAriaLabelledBy();
+  }
+
+  ngOnInit(): void {
+    if (!this.id) {
+      throw EMPTY_ID_ERROR;
+    }
+    if (!this.placeholder) {
+      throw EMPTY_PLACEHOLDER_ERROR;
+    }
```
Comment (flowg) [inline libs/ngx-dsfr/input/src/lib/input.component.ts:91]: It's `label` and not `placeholder` that should be checked here...

**Review by flowg — CHANGES_REQUESTED** [review]

File: libs/ngx-dsfr/input/src/lib/input.component.html [hunk 6]
```diff
@@ -0,0 +1,49 @@
...
+      </span>
+    </label>
+    <ng-container *ngIf="type === 'textarea'; else standardInput">
+      <textarea
+        class="fr-input"
+        [name]="name ? name : ''"
+        [id]="id"
+        [placeholder]="placeholder"
+        [attr.aria-labelledby]="ariaLabelledBy"
+        [disabled]="disabled ? 'disabled' : false"
```
[the remark this answers is not in the document]
Comment (abdelazizsakhi) [reply libs/ngx-dsfr/input/src/lib/input.component.html:16]: Nope, it is not working with `[disabled]="disabled"`

File: apps/playground/src/app/modules/components/link-module/link-module.component.ts [hunk 7]
```diff
@@ -38,12 +38,12 @@ type FormInlineTrue = {
 export class LinkModuleComponent implements OnInit, OnDestroy {
   iconAlignment: typeof ElementAlignment = ElementAlignment;
   linkSize: typeof ElementSize = ElementSize;
-  formInlineTrue: FormGroup<FormInlineTrue> | undefined;
+  formInlineTrue: FormGroup<any> | undefined;
```
Comment (flowg) [inline apps/playground/src/app/modules/components/link-module/link-module.component.ts:41]: Why did you change that ?

File: libs/ngx-dsfr/input/src/lib/input.component.html [hunk 8]
```diff
@@ -0,0 +1,49 @@
...
+        [ngClass]="{'fr-input--error': hasFailed, 'fr-input--valid': hasSucceeded}"
+        (blur)="onBlur($event)"
+        (keyUp)="onKeyup($event)"
+        (focus)="onFocus($event)"
+        [(ngModel)]="value">
+      </textarea>
+    </ng-container>
+    <ng-template #standardInput>
+      <div [ngClass]="icon ? 'fr-input-wrap ' + icon : ''">
```
[the remark this answers is not in the document]
Comment (flowg) [reply libs/ngx-dsfr/input/src/lib/input.component.html]: Not treated

File: libs/ngx-dsfr/input/src/lib/input.component.html [hunk 9]
```diff
@@ -1,49 +1,48 @@
...
         class="fr-input"
+        [type]="type"
         [name]="name ? name : ''"
         [id]="id"
         [placeholder]="placeholder"
-        [attr.aria-labelledby]="ariaLabelledBy"
-        [disabled]="disabled ? 'disabled' : false"
+        [attr.aria-describedby]="ariaDescribedby"
+        [disabled]="disabled"
         [ngClass]="{'fr-input--error': hasFailed, 'fr-input--valid': hasSucceeded}"
```
Comment (flowg) [inline libs/ngx-dsfr/input/src/lib/input.component.html:35]: For consistency, same remark here on how to handle classes depending on Component logic and Input properties

File: libs/ngx-dsfr/input/src/lib/testing/input.harness.ts [hunk 10]
```diff
@@ -0,0 +1,58 @@
...
+ * This class inherits from ComponentHarness and its purpose
+ * is to serve as a wrapper around an instance of DsfrRadioComponent
+ * within Unit Tests
+ *
+ * To see more : https://material.angular.io/cdk/test-harnesses/overview
+ */
+export class DsfrInputHarness extends ComponentHarness {
+  static hostSelector: string = 'dsfr-input';
+
+  private getInputElement: AsyncFactoryFn<TestElement> = this.locatorFor('input[type=text]');
```
[the remark this answers is not in the document]
Comment (flowg) [reply libs/ngx-dsfr/input/src/lib/testing/input.harness.ts:20]: Not treated

File: libs/ngx-dsfr/input/src/lib/testing/input.harness.ts (hunk 2 above)
Comment (flowg) [reply libs/ngx-dsfr/input/src/lib/testing/input.harness.ts:53]: Not treated

File: libs/ngx-dsfr/stepper/src/lib/stepper.component.scss [hunk 11]
```diff
@@ -1 +1 @@
- @use "@gouvfr/dsfr/dist/component/stepper/stepper.min";
+ // @use "@gouvfr/dsfr/dist/component/stepper/stepper.min";
```
Comment (flowg) [inline libs/ngx-dsfr/stepper/src/lib/stepper.component.scss:1]: Why did you change that ?

**Review by flowg — CHANGES_REQUESTED** [review]

**Pull request merged**